src/panels/lovelace/editor/hui-edit-card.ts

```diff
@@ -77,13 +93,37 @@ export class HuiEditCard extends hassLocalizeLitMixin(LitElement) {
     this._saving = false;
   }
 
+  set viewId(viewId: string) {
+    if (!viewId) {
+      return;
+    }
+    if (String(viewId) !== this._viewId) {
+      this._configValue = { format: "yaml", value: undefined };
+      this._configState = "OK";
```

Comment:
These lines feel very similar to `set cardConfig`. I wonder if we should just check this in `updated()` instead of in a setter. We can then combine the logic.